File: interf.py
import mysql.connector
import random
from datetime import datetime

DATABASE_HOST = '10.0.14.4'  
DATABASE_NAME = 'eau_pure'  
DATABASE_USER = 'root'  
DATABASE_PASSWORD = 'ieufdl'
DATABASE_PORT = '9999'  # Port mysql ouvert sur le serveur

PLUVIOMETER_SENSOR_ID = 1
LIMNIMETER_SENSOR_ID = 2
def Envois_mesures(capteur, valeur, unite, date):
    conn = None
    cursor = None
    try:
        # Conversion avant connexion, pour attraper l'erreur tôt
        valeur = float(valeur)
    except ValueError as ve:
        print(f"Erreur conversion valeur en float: {ve}")
        return  # Quitter la fonction car la valeur est invalide

    try:
        # Connexion à la BDD
        conn = mysql.connector.connect(
            host=DATABASE_HOST,
            database=DATABASE_NAME,
            user=DATABASE_USER,
            password=DATABASE_PASSWORD,
            port=DATABASE_PORT,
        )
        cursor = conn.cursor()

        # Insertion des données
        cursor.execute('''
        INSERT INTO Mesure (capteur, valeur, unite, date)
        VALUES (%s, %s, %s, %s)
        ''', (capteur, valeur, unite, date))

        conn.commit()
        print(f"Mesure insérée: {capteur}, {valeur}, {unite}, {date}")

    except mysql.connector.Error as err:
        print(f"Erreur MySQL: {err}")
    except Exception as e:
        print(f"Erreur: {e}")
    finally:
        if cursor:
            cursor.close()
        if conn and conn.is_connected():
            conn.close()


db_config = {
    'host': 'localhost',
    'user': 'root',
    'password': 'ieufdl',
    'database': 'eau_pure',
    'port': '9999'
}

File: interf.php
<?php
function connectDB() {
    $host = '10.0.14.4';
    $dbname = 'eau_pure';
    $user = 'root';
    $password = 'ieufdl';
    $port = 9999;

    $conn = new mysqli($host, $user, $password, $dbname, $port);

    if ($conn->connect_error) {
        die("Erreur de connexion : " . $conn->connect_error);
    }

    return $conn;
}

function getRivieres($conn) {
    $rivieres = [];
    $sql = "SELECT id, riviere, latitude, longitude FROM Station";
    $result = $conn->query($sql);

    if ($result && $result->num_rows > 0) {
        while ($row = $result->fetch_assoc()) {
            $rivieres[] = $row;
        }
    } else {
        $rivieres[] = ["id" => "", "riviere" => "Aucune rivière", "latitude" => "", "longitude" => ""];
    }

    return $rivieres;
}

function insertEchantillon($conn, $date, $preleveur, $technicien_id, $station_id) {
    $stmt = $conn->prepare("INSERT INTO Echantillon (date, preleveur, station_id) VALUES (?, ?, ?, ?)");
    if (!$stmt) {
        die("Erreur préparation Echantillon : " . $conn->error);
    }

    $stmt->bind_param('sisi', $date, $preleveur, $technicien_id, $station_id);

    if (!$stmt->execute()) {
        die("Erreur insertion Echantillon : " . $stmt->error);
    }

    $id = $stmt->insert_id;
    $stmt->close();
    return $id;
}

function insertAnalyse($conn, $prelevement_id, $analyses) {
    $stmt = $conn->prepare("INSERT INTO Analyse (prelevement, valeur, unite, type) VALUES (?, ?, ?, ?)");
    if (!$stmt) {
        die("Erreur préparation Analyse : " . $conn->error);
    }

    foreach ($analyses as $type => $info) {
        $stmt->bind_param('idss', $prelevement_id, $info['value'], $info['unite'], $type);
        if (!$stmt->execute()) {
            die("Erreur insertion Analyse pour $type : " . $stmt->error);
        }
    }

    $stmt->close();
}
?>
